src/types/list.rs

```diff
@@ -40,6 +40,17 @@ impl PyList {
         }
     }
 
+    /// Construct a new empty list with the given capacity.
+    pub fn with_capacity<'p>(py: Python<'p>, capacity: usize) -> &'p PyList {
+        unsafe {
+            let ptr = ffi::PyList_New(capacity as Py_ssize_t);
+            for i in 0..capacity {
+                ffi::PyList_SetItem(ptr, i as Py_ssize_t, Py_None);
```

Comment:
`Py_None` is a function so please call it like `Py_None()`

Reply:
Please use `py.None()`. `Py_None` is a singleton and you need to take care of reference counting. s. https://docs.python.org/3.7/c-api/none.html#c.Py_None

Reply:
Oh I didn't know that, thanks.

Reply:
Will do.